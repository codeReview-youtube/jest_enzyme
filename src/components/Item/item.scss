.item {
  display: flex;
  flex-direction: row;
  justify-content: space-between; 
  align-items: center;
  font-size: 1.5em;
  font-weight: 500;
  line-height: 2em;
  margin-bottom: 1em;
  border-radius: 0.5em;
  color: white;
  text-align: left;
  padding: 0.2em;
  .todo {
    font-family: 'Fira Code';
    text-transform: capitalize;
    text-align: left;
    width: 70%;
    padding-left: 0.2em;
  }
  .actions {
    width: 20%;
    border-bottom-right-radius: 0.5em;
    border-top-right-radius: 0.5em;
    .delete,
    .check {
      width: 100%;
      background-color: #96a9bf;
      border-top-left-radius: 2em;
      text-transform: capitalize;
      text-align: center;
      cursor: pointer;
    }
  }
}
