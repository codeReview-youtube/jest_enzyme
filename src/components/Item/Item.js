import React from 'react';
import './item.scss';
/**
 * @author
 * @function Item
 **/

const Item = ({ item, onClick, onDelete, onEdit }) => {
  const isCompleted = item.completed ? 'orange' : '#393B40';

  return (
    <div className='item' style={{ backgroundColor: isCompleted }}>
      <div className='todo' onClick={onEdit}>
        {item.title}
      </div>
      <div className='actions'>
        <div
          className='check'
          onClick={onClick}
          style={{ backgroundColor: isCompleted }}>
          {item.completed ? 'Done' : 'Check'}
        </div>
        {item.completed ? (
          <div
            className='delete'
            onClick={item.completed ? onDelete : () => {}}>
            delete
          </div>
        ) : (
          ''
        )}
      </div>
    </div>
  );
};

export default Item;
