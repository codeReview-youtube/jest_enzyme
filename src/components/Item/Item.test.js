/**
 * @jest-environment jsdom
 */
/* eslint-disable react/display-name */
import React from 'react';
import { shallow } from 'enzyme';
/**
 *import  Component
 */
import Item from './Item';
/**
 * props
 */
const props = {
  item: {
    id: 1,
    title: 'new Todo',
    completed: false
  },
  onClick: jest.fn(),
  onDelete: jest.fn(),
  onEdit: jest.fn()
};
/**
 * Tests: Item Component
 */
describe('Item Component', () => {
  /**
   * Snapshot:[default]
   */
  test('should match default snapshot', () => {
    expect.assertions(1);

    const wrapper = shallow(<Item {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  /**
   * Snapshot:[completed]
   */
  test('should match with item completed', () => {
    expect.assertions(1);

    const wrapper = shallow(
      <Item {...props} item={{ ...props.item, completed: true }} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
