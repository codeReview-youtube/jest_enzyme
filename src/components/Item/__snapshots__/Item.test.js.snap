// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should match default snapshot 1`] = `
<div
  className="item"
  style={
    Object {
      "backgroundColor": "#393B40",
    }
  }
>
  <div
    className="todo"
    onClick={[MockFunction]}
  >
    new Todo
  </div>
  <div
    className="actions"
  >
    <div
      className="check"
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "#393B40",
        }
      }
    >
      Check
    </div>
  </div>
</div>
`;

exports[`Item Component should match with item completed 1`] = `
<div
  className="item"
  style={
    Object {
      "backgroundColor": "orange",
    }
  }
>
  <div
    className="todo"
    onClick={[MockFunction]}
  >
    new Todo
  </div>
  <div
    className="actions"
  >
    <div
      className="check"
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
    >
      Done
    </div>
    <div
      className="delete"
      onClick={[MockFunction]}
    >
      delete
    </div>
  </div>
</div>
`;
